redbadger/pride-london-app
# Welcome Screen: Bad Connection

This PR adapts the welcome screen in the following ways:

* Hold the welcome screen if no data
* Show a bad connection message
* Retry if a user clicks on the Retry button
* Trigger animation on data load complete
* Don't show the message if the user has local storage

![2018-06-30 14 45 09](https://user-images.githubusercontent.com/719550/42125784-d38d4268-7c74-11e8-99a8-40553c83cbac.gif)

Trello: https://trello.com/c/VzYYoXoC

## Test plan
* Delete the app and re-install
* Before starting turn flight mode on
* You should see the welcome screen then the bad connection message
* Hit retry and spinner should show. After the timeout, the retry button should re-appear
* Turn off flight mode
* Hit retry and spinner should show before continuing with welcome animation
* Close and restart the app. You should see no bad connection message

## Pre-flight check-list

Before raising a pull request


## Pre-merge check-list

* [x] Link to Trello ticket/GitHub issue (If applicable)
* [x] Any risky areas identified

## Tester check-list

* [ ] Tested on multiple devices / OS versions (Test on the physical device(s) you have access to, otherwise use BrowserStack):

  * [ ] Galaxy S8 (Android 7.0/8.0)
  * [ ] Galaxy S7 (Android 6.0)
  * [ ] iPhone X (iOS 11.x)
  * [ ] iPhone 8 (11.x)
  * [ ] iPhone 7 (iOS 10.x)

  Optional:

  * [ ] Google Pixel 2
  * [ ] Galaxy S8 Plus
  * [ ] Galaxy S7 Edge
  * [ ] Galaxy S6
  * [ ] iPhone 7+
  * [ ] iPhone 6

Accessibility Testing (If applicable)

* [ ] Text-to-Voice (Android: Voice Assistant / iOS: Speak Selection)
* [ ] Large Text (=<200%)
* [ ] Landscape Screen orientation

Weak connection testing (If applicable)

* [ ] Airplane mode
* [ ] 2G/3G Network Settings
